src/common/qoi.h

```diff
@@ -0,0 +1,649 @@
+/*
```

Comment:
This file should probably live in /src/external[/qoi] (see xcf)

Reply:
Perhaps this is also logical. But I followed the analogy of placing the Permutohedral.h file in a folder next to the source files that use it (/src/iop). I don't have a strong opinion on this. I will not vigorously defend my choice, although I don't see why transferring to /src/external is better either.

Reply:
For me, there are several practical reasons to keep source not of dt provenance separate:

- easier to identify periodically what needs to be updated from upstream
- easier to audit license/IP issues

I'd go even as far as actually using as git submodule with an identifiable commit instead of plain copy when possible.

Reply:
But I am not the maintainer, so this was just a suggestion of course. 😉

Reply:
> For me, there are several practical reasons to keep source not of dt provenance separate:
> 
> * easier to identify periodically what needs to be updated from upstream
> * easier to track potential security advisories
> * easier to audit license/IP issues

Everyone has their own practices for working with upstream source code. You claim it's easier for you, but I find it hard to believe that a separate directory __for a single file__ will make it any easier in this regard. As for me, I don't use moving the code to a submodule or a separate directory __to solve the problems listed above__ for the following reasons:

- To understand whether something needs to be updated from upstream, I use a subscription to the git repository or, sometimes, as an option, mirroring the repository and "git log" in cron job. Will you mirror the upstream repo directly into the dt repository subdirectory to see what's new there? Of course not. It doesn't make sense.

- To track security issues, I subscribe to relevant newsletters. Keeping the code in a separate directory won't help at all.

- Regarding licensing issues, I can't even imagine how a separate directory for the source code file will help in this.

Reply:
I think that all `imageio_*` (and related files like qoi.h) in `src/common` should be moved into `src/imageio/` directory. But let's not do that as part of this PR... except if you insist, what do you think @victoryforce?

Reply:
> But let's not do that as part of this PR

I agree with you @TurboGit. Let's not do that as part of this PR. The proposed relocation of files will be more visible in the development history of darktable if it is done by a separate PR.

Reply:
PR merged. Let me know if you want me to handle the module relocations.

Reply:
> PR merged. Let me know if you want me to handle the module relocations.

I'll do it today.